src/MainWindow.cpp

```diff
@@ -963,20 +969,33 @@ void MainWindow::executeQuery()
     sqlWidget->finishExecution(statusMessage);
     updatePlot(sqlWidget->getModel());
 
+    connect(sqlWidget->getTableResult(), SIGNAL(clicked(QModelIndex)), this, SLOT(dataTableSelectionChanged(QModelIndex)));
+
+	connect(sqlWidget->getTableResult(), SIGNAL(doubleClicked(QModelIndex)), this,
+			SLOT(doubleClickTable(QModelIndex)));
+
     if(!modified && !wasdirty)
         db.revertToSavepoint(); // better rollback, if the logic is not enough we can tune it.
 }
 
 void MainWindow::mainTabSelected(int tabindex)
 {
+    editDock->allowEditing(false);
+
     if(tabindex == 0)
     {
         populateStructure();
     } else if(tabindex == 1) {
+        m_currentTabTableModel = m_browseTableModel;
         populateStructure();
         resetBrowser();
     } else if(tabindex == 2) {
         loadPragmas();
+    } else if(tabindex == 3) {
+        SqlExecutionArea* sqlWidget = qobject_cast<SqlExecutionArea*>(ui->tabSqlAreas->currentWidget());
+
+        if (sqlWidget)
+            m_currentTabTableModel = sqlWidget->getModel();
```

Comment:
I think we should include something like this inside the if-block:
```
dataTableSelectionChanged(sqlWidget->getTableResult()->currentIndex());
```
The idea here is that switching from the Browse Data tab to the Execute SQL tab wouldn't update the value shown in the edit cell panel even if there was a selection. This line should ensure that the contents are always updated, or so I hope.